### PR TITLE
tesseract-ocr: update to 5.3.3

### PR DESCRIPTION
**Pull request by chrysos349**
@newbluemoon
`ccextractor` was updated to the latest version, because it had to be rebuilt for tesseract-ocr-5.3.2 anyway.

@Piraty 
`arcan` was revbumped for tesseract-ocr-5.3.2 .

#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)
- I built this PR locally for these architectures (if supported. mark crossbuilds): 
  - i686
  - aarch64
  - armv7l
  - x86_64-musl
  - armv6l-musl
  - aarch64-musl

### CODE REVIEW

**Comment by chrysos349** [comment]
Test `boxa3_reg` failed in two cases (others are being checked at the moment). I guess I'll have to remove this test from `prog/Makefile.am` for `leptonica` to pass, or add `make_check=no` if there are more problematic tests to be found later.

P.S. All the tests were passed on my local machine.

**Comment by newbluemoon** [comment]
Did you test if ccextractor works with `-hardsubx`? Because it segfaulted and that was for me the show stopper to update. If I have a few spare minutes I'll try and test it. Hopefully it works now :)

**Comment by chrysos349** [comment]
@newbluemoon 

> Did you test if ccextractor works with -hardsubx? 

Good catch! I didn't. All I did was fix compilation errors. Anyway, even after adding 'tesseract 5 support', it still didn't work:
```
Job 1, 'ccextractor -hardsubx video.mp4' terminated by signal SIGSEGV (Address boundary error)
```
`ccextractor-0.94` is too problematic for my tastes. I wasted enough time as it is on it.

So I reverted back to `ccextractor-0.93`, fixed the compilation error, and added  'tesseract 5 support'. It worked fine in my tests.

I hope my work is done here.

**Comment by chrysos349** [comment]
bump

**Comment by Piraty** [comment]
by now 

`leptonica-1.82.0 -> leptonica-1.84.0`

`tesseract-ocr-4.1.1 -> tesseract-ocr-5.3.3`

File: srcpkgs/arcan/template [hunk 1]
```diff
@@ -32,6 +32,12 @@ checksum="7bf083412bc61555472877313c13116431a0a36fccbf142f97559db43b4a1475
 
 export CMAKE_GENERATOR="Unix Makefiles"
 
+case "$XBPS_TARGET_MACHINE" in
+	i686*)
+		configure_args+=" -DSSE_42_DETECTED_EXITCODE=0"
```
Comment (Piraty) [inline srcpkgs/arcan/template:37]: why?

Comment (chrysos349) [reply srcpkgs/arcan/template]: not needed anymore. removed.

**Comment by chrysos349** [comment]
`leptonica` update to `1.84.0`
`tesseract-ocr` updated to `5.3.3`
`arcan` and `ccextractor` revbumped and rebuilt for `tesseract-5.3.3`

File: srcpkgs/ccextractor/template [hunk 2]
```diff
@@ -16,7 +16,12 @@ distfiles="https://github.com/CCExtractor/${pkgname}/archive/v${version}.tar.gz"
 checksum=0e66d3e360db1b02a88271af11313ca4c9bbda1b03728e264a44c4c9f77192e3
 CFLAGS="-I${XBPS_CROSS_BASE}/usr/include/tesseract -DPNG_POWERPC_VSX_OPT=0 -fcommon"
 
+if [ "$CROSS_BUILD" ]; then
+	hostmakedepends+=" tesseract-ocr-devel"
+fi
+
 pre_configure() {
+	ln -sf libleptonica.so ${XBPS_CROSS_BASE}/usr/lib/liblept.so
```
Comment (Piraty) [inline srcpkgs/ccextractor/template:24]: why?

Comment (chrysos349) [reply srcpkgs/ccextractor/template]: done

**Pull request merged**